### ISSUE TITLE
Version v37.9.2 breaks ESLint when using `jsdoc/no-restricted-syntax` rule

### ISSUE THREAD

**Issue by macku** [gajus/eslint-plugin-jsdoc#842]
After trying to upgrade `eslint-plugin-jsdoc` to the latest version  37.9.2, our CI started failing with an ESLint error.
Change: https://bitbucket.org/atlassian/atlassian-browser-devtools/pull-requests/201

The failure is related to a custom check defined for `jsdoc/no-restricted-syntax` rule I think.

After removing the custom `no-restricted-syntax` rule from the `eslintrc` file, the ESLint is no longer breaking.

## Expected behavior
The ESLint should work without an error.

## Actual behavior
ESLint is throwing an error:

```
Oops! Something went wrong! :(
ESLint: 8.9.0
Error: Unexpected type: 'JsdocTypeVariadic'.
Occurred while linting /opt/atlassian/pipelines/agent/build/src/common/sentry/sentry.js:41
  Rule: "jsdoc/no-restricted-syntax"
    at Object.parseInfix (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:1098:31)
    at keyValueParslet (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:467:40)
    at Parser.tryParslets (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:368:32)
    at Parser.parseInfixIntermediateType (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:391:34)
    at Parser.parseIntermediateType (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:384:25)
    at Object.parsePrefix (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:530:35)
    at parenthesisParslet (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:460:40)
    at Parser.tryParslets (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:368:32)
    at Parser.parseIntermediateType (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:380:33)
    at Parser.parseType (/opt/atlassian/pipelines/agent/build/node_modules/jsdoc-type-pratt-parser/dist/index.js:361:42)
```

source: https://bitbucket.org/atlassian/atlassian-browser-devtools/addon/pipelines/home#!/results/876/steps/%7B97ecf3dc-a9d3-48c4-9e72-fdd5c2ffabb4%7D

## ESLint Config

```js
    'jsdoc/no-restricted-syntax': [
      'error',
      {
        contexts: [
          {
            comment: 'JsdocBlock:has(JsdocTag[tag="type"]:has([value=/FunctionComponent/]))',
            context: 'any',
            message: 'The `FunctionComponent` type is not allowed. Please use `FC` instead.',
          },
        ],
      },
    ],
```
source: https://bitbucket.org/atlassian/atlassian-browser-devtools/src/master/.eslintrc.js

## ESLint sample

```js
/**
 * @param {(...args: any[]) => any} fn
 * @returns {(...args: any[]) => any}
 */
```
source: https://bitbucket.org/atlassian/atlassian-browser-devtools/src/6260795c608bdf6f540b72d0f8de453b320d6fcd/src/common/sentry/sentry.js#lines-35:40

## Environment

- Node version: 16
- ESLint version 8.9.0
- `eslint-plugin-jsdoc` version: 37.9.2
